src/prefs/DirectoriesPrefs.cpp

```diff
@@ -418,6 +418,11 @@ bool DirectoriesPrefs::Validate()
    }
    else {
       /* If the directory already exists, make sure it is writable */
+      if (!FileNames::WritableLocationCheck(mTempText->GetValue()) ||
+          !FileNames::WritableLocationCheck(mMacrosText->GetValue()))
+      {
+          return false;
+      }
```

Comment:
I understand, this will prevent closing of the Preferences dialog, so long as either path is bad.